tests/test_serializer.py

```diff
@@ -31,7 +32,8 @@ def test_invalid_serializer(self):
         serializer = self.Serializer(data={'char': 'abc'})
         assert not serializer.is_valid()
         assert serializer.validated_data == {}
-        assert serializer.errors == {'integer': ['This field is required.']}
+        assert serializer.errors['integer'].detail == ['This field is required.']
```

Comment:
I could change the definition of [`errors`](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/serializers.py#L247) to return the initial type.
Would that be enough ?